### PR TITLE
[GSoC] Added TF and PyTorch segmentation conversion cases

### PR DESCRIPTION
**Pull request by LupusSanctus**
GSoC project: "Write a tutorial about training a network in TF or PyTorch and then running it in OpenCV"

**Part 1.** [Classification case](https://github.com/opencv/opencv/pull/17604)
**Part 2.** Segmentation case:
* TensorFlow DNN (FCN example) => DNN conversion => OpenCV [Net](https://docs.opencv.org/4.3.0/db/d30/classcv_1_1dnn_1_1Net.html) object
* PyTorch DNN (FCN example) => DNN conversion => OpenCV [Net](https://docs.opencv.org/4.3.0/db/d30/classcv_1_1dnn_1_1Net.html) object
* Evaluation

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

force_builders_only=Docs

### CODE REVIEW

**Comment by dkurt** [comment]
It's not clear for me what this PR proposes. Can you summarize in the description?

File: samples/python/tutorial_code/dnn/dnn_conversion/common/utils.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import tensorflow as tf
+import torch
+
+SEED_VAL = 42
+DNN_LIB = "DNN"
+# common path for model savings
+MODEL_PATH_ROOT = os.path.join(DNN_LIB.lower() + "_conversion", "{}/models")
+
+
+def make_dir(dir_to_create):
```
Comment (asmorkalov) [inline samples/python/tutorial_code/dnn/dnn_conversion/common/utils.py:17]: you can replace the function with `os.path.makedirs(dir_to_create, exist_ok=True)`

File: samples/python/tutorial_code/dnn/dnn_conversion/common/utils.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+def plot_acc(accuracies_list, experiment_name):
+    plt.figure(figsize=[8, 6])
+    plt.plot(accuracies_list[:, 0], "r", linewidth=2.5, label="Original Model")
+    plt.plot(accuracies_list[:, 1], "b", linewidth=2.5, label="Converted DNN Model")
+    plt.xlabel("Iterations ", fontsize=15)
+    plt.ylabel("Accuracy", fontsize=15)
+    plt.title(experiment_name, fontsize=15)
+    plt.legend()
+    plt.savefig(experiment_name.replace(" ", "_") + ".png", bbox_inches="tight")
+    pass
```
Comment (asmorkalov) [inline samples/python/tutorial_code/dnn/dnn_conversion/common/utils.py:44]: `pass` is redundant here.

**Comment by asmorkalov** [comment]
@LupusSanctus Friendly reminder.

**Comment by asmorkalov** [comment]
@LupusSanctus the first part has been merged. Please re-base the PR and fix CI-reported issues.

**Review by alalek — APPROVED** [review]

**Comment by ethanc8** [comment]
@LupusSanctus could you please provide the images linked to the tutorials and also the full self-contained segmentation code, especially the implementations of `get_colored_mask` and `get_legend`? It's hard to understand your tutorials when there are links to nonexistent images, the referenced sample code file doesn't share any code with the tutorial, and the tutorial doesn't provide the implementations of important functions required to see if the code has been actually working.
